### ISSUE TITLE
[BUG] Rancher provisioned RKE1 clusters stay in updating state post rancher upgrade

### ISSUE THREAD

**Issue by anupama2501** [rancher/rancher#40126]
**Rancher Server Setup**
- Rancher version: `v2.5.16` upgraded to `v2.6.4`
- Installation option (Docker install/Helm Chart): RKE1, RKE1 version: `1.2.22`, k8s version: `v1.20.15`, 3 nodes, all roles
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details: Self-signed

**Information about the Cluster**
- Kubernetes version: `v1.20.15`
- Cluster Type (Local/Downstream): Downstream custom 3 worker, 1 etcd, 1 cp
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? Admin
  - If custom, define the set of permissions: 

**Describe the bug**
After upgrading the rancher server from the version 2.5.16 to the version 2.6.4, the cluster is stuck in `Unavailable` state and never comes back up active. 

**To Reproduce**
1. Install rancher on v2.5.16 with the above mentioned spec
2. As an admin, create a downstream cluster with 3 worker, 1 etcd and 1 cp node
3. Upgrade rancher server to v2.6.4 by running the helm command:
```
helm upgrade --install rancher rancher-latest/rancher --version 2.6.4 --namespace cattle-system --set hostname=<hostname>
```
4. Rancher server comes back up active and upgraded to 2.6.4

**Result**
Downstream rke1 cluster stays in `Unavailable` state and never comes back up active.

**Expected Result**
Expected downstream to come back up active.

**Additional context**
Noticed that from the downstream cluster logs, the pod `cattle-cluster-agent` restarted 5 times and the following logs are seen in the rancher pod logs: 
```


2023/01/10 18:30:32 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-jxppf: ClusterUnavailable 503: cluster not found, requeuing
--
Tue, Jan 10 2023 10:30:49 am | 2023/01/10 18:30:49 [ERROR] error syncing 'c-jxppf': handler cluster-deploy: cluster context c-jxppf is unavaiblable, requeuing
Tue, Jan 10 2023 10:32:32 am | 2023/01/10 18:32:32 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-jxppf: ClusterUnavailable 503: cluster not found, requeuing
Tue, Jan 10 2023 10:32:49 am | 2023/01/10 18:32:49 [ERROR] error syncing 'c-jxppf': handler cluster-deploy: cluster context c-jxppf is unavaiblable, requeuing
Tue, Jan 10 2023 10:34:32 am | 2023/01/10 18:34:32 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-jxppf: ClusterUnavailable 503: cluster not found, requeuing
Tue, Jan 10 2023 10:34:49 am | 2023/01/10 18:34:49 [ERROR] error syncing 'c-jxppf': handler cluster-deploy: cluster context c-jxppf is unavaiblable, requeuing
Tue, Jan 10 2023 10:36:32 am | 2023/01/10 18:36:32 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-jxppf: ClusterUnavailable 503: cluster not found, requeuing
Tue, Jan 10 2023 10:36:49 am | 2023/01/10 18:36:49 [ERROR] error syncing 'c-jxppf': handler cluster-deploy: cluster context c-jxppf is unavaiblable, requeuing
Tue, Jan 10 2023 10:38:32 am | 2023/01/10 18:38:32 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-jxppf: ClusterUnavailable 503: cluster not found, requeuing
Tue, Jan 10 2023 10:38:49 am | 2023/01/10 18:38:49 [ERROR] error syncing 'c-jxppf': handler cluster-deploy: cluster context c-jxppf is unavaiblable, requeuing
Tue, Jan 10 2023 10:40:32 am | 2023/01/10 18:40:32 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-jxppf: ClusterUnavailable 503: cluster not found, requeuing
Tue, Jan 10 2023 10:40:49 am | 2023/01/10 18:40:49 [ERROR] error syncing 'c-jxppf': handler cluster-deploy: cluster context c-jxppf is unavaiblable, requeuing
Tue, Jan 10 2023 10:42:32 am | 2023/01/10 18:42:32 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-jxppf: ClusterUnavailable 503: cluster not found, requeuing
Tue, Jan 10 2023 10:42:49 am | 2023/01/10 18:42:49 [ERROR] error syncing 'c-jxppf': handler cluster-deploy: cluster context c-jxppf is unavaiblable, requeuing
Tue, Jan 10 2023 10:44:32 am | 2023/01/10 18:44:32 [ERROR] error syncing '_all_': handler user-controllers-controller: failed to start user controllers for cluster c-jxppf: ClusterUnavailable 503: cluster not found, requeuing
```

Not seeing the issue when upgrading from 2.5.16 >> 2.6.9 or 2.5.16 >> 2.7.0

**Comment by deniseschannon** [comment]
Considering the dependencies within Rancher may downgrade in this use case, we have been having conversations about how this upgrade path (later 2.5.x versions to early 2.6.x versions) will not be supported. 

Looking at when the dates of 2.5.16 was released, could we do 1 sanity test with the 2.6.7 release which was at the same timeframe as 2.5. 

@anupama2501 Can you test 2.5.16 -> 2.6.7 to see if we still have this issue or not? If this doesn't happen in that use case, then I'd say this is due to a downgrade of dependencies.

**Comment by anupama2501** [comment]
Validated that the downstream RKE1 custom clusters come back up active after upgrading rancher server from 2.5.16 >> 2.6.7

Steps:

- Install rancher on v2.5.16 on an RKE1 cluster, version: 1.2.22, k8s version: v1.20.15, 3 nodes, all roles
- As an admin create a standard user - user1
- create a downstream cluster with 3 worker, 1 etcd and 1 cp node as user1
- Upgrade rancher server to v2.6.7 by running the helm command:
- helm upgrade --install rancher rancher-latest/rancher --version 2.6.7 --namespace cattle-system --set hostname=<hostname>
- Rancher server comes back up active and upgraded to 2.6.7
- Downstream rke1 cluster  comes back up active. with the cluster-agent upgraded to `2.6.7`
